src/App.js

```diff
@@ -1,28 +1,55 @@
 import './App.css'
-import Oggetti from './Components/OggettiComponents/Oggetti'
-import AllCategories from './Components/Filter/AllCategories'
-import Cart from './Components/Cart/Cart'
+import React, { useState, useEffect } from 'react'
+
 import Header from './Components/HeaderContent'
+import Shopping from './pages/Shopping'
+import Login from './pages/Login'
+import Register from './pages/Register'
+import Logout from './pages/Logout'
+
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
+import { checkShowMe } from './utils/api'
 
 function App() {
-  return (
-    <div>
-      <header>
-        <nav className='header_content'>
-          <Header />
-        </nav>
-      </header>
+  const [user, setUser] = useState(undefined);
 
-      <div className='aside'>
-        <AllCategories />
-        <Cart />
-      </div>
+  useEffect(() => {
+    async function getLoggedInUser() {
+      const user = await checkShowMe()
+      if (user) {
+        setUser(user)
+      }
+    }
+    getLoggedInUser()
+  }, [])
 
-      <section>
-        <div className='oggetti_container'>
-          <Oggetti />
-        </div>
-      </section>
+  function resetUser(wasLogoutSuccessful) {
+    if (wasLogoutSuccessful) {
+      setUser(undefined)
+    }
+  }
+
+  return (
+    <div>
+      <Router>
+        <header>
+          <nav className='header_content'>
+            <Header username={user?.name} />
```

Comment:
Here I'm using ["optional chaining”](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) syntax:

`user?.name` means if `user` exists and is an object with the `name` property, then get the value of that `name` property, but if it doesn't exist, then just return undefined.

It's basically a shorter way of saying:

```js
if (user && user.name) {
  return user.name
} else {
  return undefined
}
```

It's also very helpful if we have a nested object, since then we can just do something like: `account?.user?.fullName?.firstName` instead of having to check if each part of that object exists and is defined. That's if we had an object like: 

```js
{
  account: {
    user: {
      fullName: {
        firstName: "blah"
        lastName: undefined
     }
  }
}
```